seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveTypeMapper.java

```diff
@@ -0,0 +1,120 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+public class HiveTypeMapper implements JdbcDialectTypeMapper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HiveTypeMapper.class);
+
+    // reference https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types
+
+    // Numeric Types
+    private static final String HIVE_TINYINT = "TINYINT";
+    private static final String HIVE_SMALLINT = "SMALLINT";
+    private static final String HIVE_INT = "INT";
+    private static final String HIVE_INTEGER = "INTEGER";
+    private static final String HIVE_BIGINT = "BIGINT";
+    private static final String HIVE_FLOAT = "FLOAT";
+    private static final String HIVE_DOUBLE = "DOUBLE";
+    private static final String HIVE_DOUBLE_PRECISION = "DOUBLE PRECISION";
+    private static final String HIVE_DECIMAL = "DECIMAL";
+    private static final String HIVE_NUMERIC = "NUMERIC";
+    // Date/Time Types
+    private static final String HIVE_TIMESTAMP = "TIMESTAMP";
+    private static final String HIVE_DATE = "DATE";
+    private static final String HIVE_INTERVAL = "INTERVAL";
+    // String Types
+    private static final String HIVE_STRING = "STRING";
+    private static final String HIVE_VARCHAR = "VARCHAR";
+    private static final String HIVE_CHAR = "CHAR";
+    // Misc Types
+    private static final String HIVE_BOOLEAN = "BOOLEAN";
+    private static final String HIVE_BINARY = "BINARY";
+    // Complex Types
+    private static final String HIVE_ARRAY = "ARRAY";
+    private static final String HIVE_MAP = "MAP";
+    // float
+    private static final String HIVE_STRUCT = "STRUCT";
+    private static final String HIVE_UNIONTYPE = "UNIONTYPE";
+
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex)
+            throws SQLException {
+        String columnType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        int precision = metadata.getPrecision(colIndex);
+        switch (columnType) {
+            case HIVE_TINYINT:
+            case HIVE_SMALLINT:
+                return BasicType.SHORT_TYPE;
+            case HIVE_INT:
+            case HIVE_INTEGER:
+                return BasicType.INT_TYPE;
+            case HIVE_BIGINT:
+                return BasicType.LONG_TYPE;
+            case HIVE_FLOAT:
+                return BasicType.FLOAT_TYPE;
+            case HIVE_DOUBLE:
+            case HIVE_DOUBLE_PRECISION:
+                return BasicType.DOUBLE_TYPE;
+            case HIVE_DECIMAL:
+            case HIVE_NUMERIC:
+                if (precision > 0) {
+                    return new DecimalType(precision, metadata.getScale(colIndex));
+                }
+                LOG.warn("decimal did define precision,scale, will be Decimal(38,18)");
+                return new DecimalType(38, 18);
```

Comment:
```
                if (scale == 0) {
                    if (precision <= 9) {
                        return BasicType.INT_TYPE;
                    } else if (precision <= 18) {
                        return BasicType.LONG_TYPE;
                    } else if (precision <= 38) {
                        return new DecimalType(38, 0);
                    }
                }
                return new DecimalType(38, 18);
```

Reply:
I think it should maintain similar data types

Reply:
> I think it should maintain similar data types

You are correct.